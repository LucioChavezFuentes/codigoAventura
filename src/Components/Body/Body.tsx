import React from 'react';
import {BrowserRouter , Route, Switch} from 'react-router-dom';
import { CSSTransition, TransitionGroup} from 'react-transition-group';
