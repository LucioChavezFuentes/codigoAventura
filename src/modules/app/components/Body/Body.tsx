import React from 'react';
import { Route, Switch} from 'react-router-dom';
import { CSSTransition, TransitionGroup} from 'react-transition-group';
import {Home , About, Contact, Error} from '../Routes';
import './Body.scss';


const Body:React.FC = () => {
    
    return(
      <div>

      
        <Route 
// tslint:disable-next-line: jsx-no-multiline-js
              render={({location}) => (
                <TransitionGroup>

                
                  <CSSTransition timeout={300} classNames='fade' key={location.key}>
                    
                    <Switch location={location}>
                      
                      <Route path="/" component={Home} exact  />

                      <Route path="/about" component={About}  />

                      <Route path="/contact" component={Contact}  />

                      <Route component={Error}/>

                    </Switch>
                  </CSSTransition>
                </TransitionGroup>
                )} 
          />
        
      </div>
    ) 
}

export default Body;

