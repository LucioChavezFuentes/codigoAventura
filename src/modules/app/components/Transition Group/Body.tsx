import React from 'react';
import { Route, Switch} from 'react-router-dom';
import { CSSTransition, TransitionGroup} from 'react-transition-group';
import Home from '../Routes/Home/Home';
import About from '../Routes/About/About';
import Contact from '../Routes/Contact/Contact';
import Error from '../Routes/Error/Error';
import './Body.scss';


const Body:React.FC = () => {
    
    return(
      <div>

      
        <Route 
// tslint:disable-next-line: jsx-no-multiline-js
              render={({location}) => (
                <TransitionGroup>

                
                  <CSSTransition timeout={300} classNames='fade' key={location.key}>
                    
                    <Switch location={location}>
                      
                      <Route path="/" component={Home} exact  />

                      <Route path="/about" component={About}  />

                      <Route path="/contact" component={Contact}  />

                      <Route component={Error}/>

                    </Switch>
                  </CSSTransition>
                </TransitionGroup>
                )} 
          />
        
      </div>
    ) 
}

export default Body;

