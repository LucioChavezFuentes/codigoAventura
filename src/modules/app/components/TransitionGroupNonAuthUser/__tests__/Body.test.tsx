import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { CSSTransition } from 'react-transition-group';
import Body from '../TransitionGroupNonAuthUser';
import { render, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders body without crashing', () => {
  const componentBody =  renderer
  .create(<BrowserRouter><Body/></BrowserRouter>)
  .toJSON();
expect(componentBody).toMatchSnapshot();
  //expect(componentBody.contains(<Body/>)).toEqual(true);
});


//The following comment is a code example about testing CSS Transitions

/*

interface fadeProps {
 in: boolean
}

const Fade : React.FC<fadeProps> = ({ children, ...props }) => {
  return (
    <CSSTransition {...props} timeout={300} classNames="fade">
      {children}
    </CSSTransition>
  )
}

interface hiddenMesageProps {
  initialShow: boolean
}

interface hiddenMessageState {
  show: boolean
}

class HiddenMessage extends React.Component<hiddenMesageProps,hiddenMessageState> {
  state : hiddenMessageState = { show: this.props.initialShow || false }
  
  toggle = () => {
    this.setState(({ show }) => ({ show: !show }))
  }
  render() {
    return (
      <div>
        <button onClick={this.toggle}>Toggle</button>
        <Fade in={this.state.show}>
          <div>Hello world</div>
        </Fade>
      </div>
    )
  }
}


afterEach(cleanup)

jest.mock('react-transition-group', () => {
  const FakeTransition = jest.fn(({ children }) => children)
  const FakeCSSTransition = jest.fn(props =>
    props.in ? <FakeTransition>{props.children}</FakeTransition> : null
  )
  return { CSSTransition: FakeCSSTransition, Transition: FakeTransition }
})

test('you can mock things with jest.mock', () => {
  const { getByText, queryByText } = render(
    <Body />
  )
  expect(getByText('Hello world')).toBeTruthy() // we just care it exists
  // hide the message
  fireEvent.click(getByText('Toggle'))
  // in the real world, the CSSTransition component would take some time
  // before finishing the animation which would actually hide the message.
  // So we've mocked it out for our tests to make it happen instantly
  expect(queryByText('Hello World')).toBeNull() // we just care it doesn't exist
})

*/