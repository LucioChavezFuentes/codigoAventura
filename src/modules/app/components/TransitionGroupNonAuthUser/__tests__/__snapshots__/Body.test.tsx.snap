// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the landing page without crashing 1`] = `
<div>
  <div
    className="page"
  >
    <div>
      <div
        className="landing-page"
      >
        <div
          className="header"
        >
          <div
            className="header-container"
          >
            <div
              className="leftSection"
            >
              <a
                aria-current="page"
                className="active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                <button
                  className="btn btn-info"
                  disabled={false}
                  type="button"
                >
                  Código Aventura
                </button>
              </a>
            </div>
            <div
              className="rightSection"
            >
              <div
                className="signIn"
              >
                <a
                  aria-current={null}
                  href="/signin"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "black",
                      "fontWeight": "bold",
                    }
                  }
                >
                  Iniciar Sesión
                </a>
              </div>
              <div
                className="signUpLink"
              >
                <a
                  aria-current={null}
                  href="/signup"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Resgístrate
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="landing-content"
        >
          <div
            className="greeting"
          >
            <div>
              <p>
                !Cualquiera puede programar!
              </p>
              <p>
                Es más que una carrera... Es una aventura.
              </p>
            </div>
            <div>
              <p>
                Y aprender aquí está papita.
              </p>
            </div>
          </div>
          <div
            className="registration"
          >
            <div>
              <form
                className="formComponent"
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    className="emailWord form-label"
                    htmlFor="inputEmail"
                  >
                    Email o Correo Electrónico
                  </label>
                  <input
                    className="form-control form-control-lg"
                    id="inputEmail"
                    name="email"
                    onChange={[Function]}
                    required={true}
                    type="email"
                    value=""
                  />
                  <div
                    className="invalid-feedback"
                  >
                    Can't be blank
                  </div>
                </div>
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="inputPasswordOne"
                  >
                    Contraseña
                  </label>
                  <input
                    className="form-control form-control-lg"
                    id="inputPasswordOne"
                    name="passwordOne"
                    onChange={[Function]}
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="inputPasswordTwo"
                  >
                    Confirma tu contraseña.
                  </label>
                  <input
                    className="form-control form-control-lg"
                    id="inputPasswordTwo"
                    name="passwordTwo"
                    onChange={[Function]}
                    required={true}
                    type="password"
                    value=""
                  />
                  <div
                    className="invalid-feedback"
                  >
                    La contraseña tiene que ser de al menos 6 caracteres.
                  </div>
                </div>
                <button
                  className="submit-btn btn btn-primary btn-block"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
